compiler/rustc_type_ir/src/binder.rs

```diff
@@ -276,15 +276,17 @@ pub struct ValidateBoundVars<I: Interner> {
     binder_index: ty::DebruijnIndex,
     // We may encounter the same variable at different levels of binding, so
     // this can't just be `Ty`
-    visited: SsoHashSet<(ty::DebruijnIndex, I::Ty)>,
+    visited_tys: SsoHashSet<(ty::DebruijnIndex, I::Ty)>,
+    visited_consts: SsoHashSet<(ty::DebruijnIndex, I::Const)>,
 }
 
 impl<I: Interner> ValidateBoundVars<I> {
     pub fn new(bound_vars: I::BoundVarKinds) -> Self {
         ValidateBoundVars {
             bound_vars,
             binder_index: ty::INNERMOST,
-            visited: SsoHashSet::default(),
+            visited_tys: SsoHashSet::default(),
```

Comment:
this cache is only for perf. I think we generally only cache tys as visiting consts has to go through a ty at some point.

We have a bunch of `Ty` based caches in folders regardless of whether they treat tys and consts the same

Reply:
yeah, remove the caching in `visit_const`